mesonbuild/backend/ninjabackend.py

```diff
@@ -1944,6 +1945,7 @@ def generate_link(self, target, outfile, outname, obj_list, linker, extra_args=[
         commands = []
         if not isinstance(target, build.StaticLibrary):
             commands += self.build.get_global_link_args(linker)
+            commands += self.build.get_project_link_args(linker, target.subproject)
```

Comment:
At least `-L` arguments should be the other way around. The first one is used so for those project link args should be before global ones. But what about other linker arguments?  I _think_ that for other ones the last one is the deciding one. Maybe we should just stick to this order and fix things if they actually break in the wild?